src/main/kotlin/org/koitharu/kotatsu/parsers/site/madara/en/IsekaiScan.kt

```diff
@@ -0,0 +1,172 @@
+package org.koitharu.kotatsu.parsers.site.madara.pt
+
+
+import org.jsoup.nodes.Document
+import org.koitharu.kotatsu.parsers.MangaLoaderContext
+import org.koitharu.kotatsu.parsers.MangaSourceParser
+import org.koitharu.kotatsu.parsers.model.Manga
+import org.koitharu.kotatsu.parsers.model.MangaChapter
+import org.koitharu.kotatsu.parsers.model.MangaPage
+import org.koitharu.kotatsu.parsers.model.MangaSource
+import org.koitharu.kotatsu.parsers.model.MangaState
+import org.koitharu.kotatsu.parsers.model.MangaTag
+import org.koitharu.kotatsu.parsers.model.SortOrder
+import org.koitharu.kotatsu.parsers.site.madara.MadaraParser
+import org.koitharu.kotatsu.parsers.util.attrAsRelativeUrlOrNull
+import org.koitharu.kotatsu.parsers.util.domain
+import org.koitharu.kotatsu.parsers.util.generateUid
+import org.koitharu.kotatsu.parsers.util.host
+import org.koitharu.kotatsu.parsers.util.mapChapters
+import org.koitharu.kotatsu.parsers.util.mapNotNullToSet
+import org.koitharu.kotatsu.parsers.util.parseFailed
+import org.koitharu.kotatsu.parsers.util.parseHtml
+import org.koitharu.kotatsu.parsers.util.removeSuffix
+import org.koitharu.kotatsu.parsers.util.toAbsoluteUrl
+import org.koitharu.kotatsu.parsers.util.toTitleCase
+import org.koitharu.kotatsu.parsers.util.urlEncoded
+import java.text.SimpleDateFormat
+import java.util.EnumSet
+
+@MangaSourceParser("ISEKAISCAN", "Isekai Scan", "en")
+internal class IsekaiScan(context: MangaLoaderContext) :
+	MadaraParser(context, MangaSource.ISEKAISCAN, "isekaiscan.top", 16) {
+
+	override val tagPrefix = "mangas/"
+	override val datePattern = "MMMM d, HH:mm"
+
+	override val sortOrders: Set<SortOrder> = EnumSet.of(
+		SortOrder.POPULARITY,
+		SortOrder.UPDATED,
+	)
+
+	override suspend fun getListPage(
+		page: Int,
+		query: String?,
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder,
+	): List<Manga> {
+		val url = buildString {
+			append("https://")
+			append(domain)
+			val pages = page + 1
+
+			when {
+				!query.isNullOrEmpty() -> {
+
+					append("/?search=")
+					append(query.urlEncoded())
+					append("&page=")
+					append(pages.toString())
+					append("&post_type=wp-manga")
+				}
+
+				!tags.isNullOrEmpty() -> {
+					append("/mangas/")
+					for (tag in tags) {
+						append(tag.key)
+					}
+
+					append("?orderby=2&page=")
+					append(pages.toString())
+
+				}
+
+				else -> {
+
+					if (sortOrder == SortOrder.POPULARITY) {
+						append("/popular-manga")
+					}
+					if (sortOrder == SortOrder.UPDATED) {
+						append("/latest-manga")
+					}
+					append("?page=")
+					append(pages.toString())
+				}
+			}
+		}
+		val doc = webClient.httpGet(url).parseHtml()
+		return doc.select("div.row.c-tabs-item__content").ifEmpty {
+			doc.select("div.page-item-detail.manga")
+		}.map { div ->
+			val href = div.selectFirst("a")?.attrAsRelativeUrlOrNull("href") ?: div.parseFailed("Link not found")
```

Comment:
```suggestion
			val href = div.selectFirstOrThrow("a").attrAsRelativeUrl("href")
```

`parseFailed` do not provide so informative error messages so we have to use it only if there are no other ways